server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -145,6 +145,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion WAIT_FOR_CLUSTER_STATE_IN_RECOVERY_ADDED = def(8_502_00_0);
     public static final TransportVersion RECOVERY_COMMIT_TOO_NEW_EXCEPTION_ADDED = def(8_503_00_0);
     public static final TransportVersion NODE_INFO_COMPONENT_VERSIONS_ADDED = def(8_504_00_0);
+    public static final TransportVersion LOG_TOO_MANY_BUCKETS_ERROR = def(8_505_00_0);
```

Comment:
I think for the logging aspect of this change we don't need this?
The exception is thrown in the same thread that logs this.
Given that the `TooManyBucketsException` does (de)serialise other properties, it makes sense to also do that for the new property.

Maybe rename this `BUCKET_COUNT_ADDED_TO_TOO_MANY_BUCKETS_EXCEPTION`?